### PR TITLE
Add <codemirror-editor>

### PR DESCRIPTION
**Pull request by justinfagnani**

### CODE REVIEW

File: src/lib/codemirror-editor.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+   */
+  @property()
+  type: 'js' | 'html' | 'css' | undefined;
+
+  render() {
+    return html`${this._editorView?.dom}`;
+  }
+
+  update(changedProperties: PropertyValues) {
+    if (changedProperties.has('value') || changedProperties.has('src')) {
```
Comment (e111077) [inline src/lib/codemirror-editor.ts:92]: Would there be any practical use as to check that this.value and this.src are not being unset?

Comment (justinfagnani) [reply src/lib/codemirror-editor.ts:92]: It's fine if they're unset, we just need to make a new view if the content is changed from the outside.

Actually, we might not always need to, but I'm still learning the CodeMirror6 API. This works for the limited use case here.

File: src/lib/codemirror-editor.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
+/**
+ * @license
+ * Copyright (c) 2019 The Polymer Project Authors. All rights reserved.
```
Comment (e111077) [inline src/lib/codemirror-editor.ts:3]: 2020

**Review by aomarks — APPROVED** [review]

File: src/lib/codemirror-editor.ts [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+      }),
+      root: this.shadowRoot!,
+    });
+    this._editorView = view;
+    this.requestUpdate();
+  }
+
+  private _getLanguagePlugins() {
+    switch (this.type) {
+      case 'js':
```
Comment (e111077) [inline src/lib/codemirror-editor.ts:136]: is TS not a built in one or is is jslang?

**Review by e111077 — CHANGES_REQUESTED** [review]
nits and questions

**Review by e111077 — APPROVED** [review]
preapproval

File: src/lib/codemirror-editor.ts (hunk 2 above)
Comment (justinfagnani) [reply src/lib/codemirror-editor.ts:3]: done

File: src/lib/codemirror-editor.ts (hunk 3 above)
Comment (justinfagnani) [reply src/lib/codemirror-editor.ts:136]: I think TS just started landing as a "variant" of JS. I'll look into it soon.

**Pull request merged**